lib/etcdv3.rb

```diff
@@ -21,8 +21,9 @@ class Etcdv3
   attr_reader :conn, :credentials, :options
   DEFAULT_TIMEOUT = 120
 
-  def initialize(options = {})
+  def initialize(allow_reconnect: true, **options)
     @options = options
+    @options[:allow_reconnect] = allow_reconnect
```

Comment:
```
[1] pry(main)> opts = { test: false }
=> {:test=>false}
[2] pry(main)> opts[:test] ||= true
=> true
[3] pry(main)> opts[:test]
=> true
```

I guess I can use something like

```
@options[:allow_reconnect] = true unless @options.key?(:allow_reconnect)
```

But I thought it would be nice to have the default value visible clearly on the `def ...` line.

Reply:
Ahh, yeah...  

Maybe `.fetch` is the answer here?  
https://apidock.com/ruby/Hash/fetch

```ruby
irb(main):028:0* opts = { allow_reconnect: false }
=> {:allow_reconnect=>false}
irb(main):029:0>
irb(main):030:0* opts.fetch(:allow_reconnect, true)
=> false
irb(main):031:0> opts.fetch(:allow_reconnectttttt, true)
=> true
```

Reply:
So many ways to do anything in ruby... Yeah, fetch looks like best option :)